framework/web/Request.php

```diff
@@ -187,6 +188,24 @@ public function resolve()
             }
             return [$route, $this->getQueryParams()];
         } else {
+            if (Yii::$app->getUrlManager()->enablePrettyUrl && !empty(Yii::$app->getUrlManager()->suffix)) {
+                $pathInfo = $this->getPathInfo();
+                $suffix = Yii::$app->getUrlManager()->suffix;
```

Comment:
I see now that both of our solutions, miss out the opportunity that an Url rule can have its own suffix, which is ignored now.